resolver/src/dns_lru.rs

```diff
@@ -35,11 +35,43 @@ impl LruValue {
 }
 
 #[derive(Debug)]
-pub(crate) struct DnsLru(LruCache<Query, LruValue>);
+pub(crate) struct DnsLru {
+    cache: LruCache<Query, LruValue>,
+    /// An optional minimum TTL value for positive responses.
+    ///
+    /// Positive responses with TTLs under `min_positive_ttl` will use
+    /// `` instead.
+    min_positive_ttl: Duration,
+    /// An optional minimum TTL value for negative (`NXDOMAIN`) responses.
+    ///
+    /// `NXDOMAIN` responses with TTLs under `min_negative_ttl` will use
+    /// `min_negative_ttl` instead.
+    min_negative_ttl: Duration,
+}
+
+#[derive(Copy, Clone, Debug, Default)]
+pub(crate) struct TtlConfig {
```

Comment:
This looks good. I there were discussions of the max TTL as well. Are you planning ot make those changes/?

Reply:
Yeah, I was going to in a subsequent PR, but I can do it as part of this branch if you prefer!